packages/concerto-tools/test/codegen/fromcto/odata/odatavisitor.js

```diff
@@ -0,0 +1,613 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const chai = require('chai');
+chai.should();
+const sinon = require('sinon');
+
+const ODataVisitor = require('../../../../lib/codegen/fromcto/odata/odatavisitor.js');
+
+const ClassDeclaration = require('@accordproject/concerto-core').ClassDeclaration;
+const EnumDeclaration = require('@accordproject/concerto-core').EnumDeclaration;
+const EnumValueDeclaration = require('@accordproject/concerto-core').EnumValueDeclaration;
+const Field = require('@accordproject/concerto-core').Field;
+const ModelFile = require('@accordproject/concerto-core').ModelFile;
+const ModelManager = require('@accordproject/concerto-core').ModelManager;
+const RelationshipDeclaration = require('@accordproject/concerto-core').RelationshipDeclaration;
+const Decorator = require('@accordproject/concerto-core').Decorator;
+const FileWriter = require('@accordproject/concerto-util').FileWriter;
+
+describe('ODataVisitor', function () {
```

Comment:
I wonder how easy it would be to test the validity of the output with an OData parser? Or validate the output against sample data for example?

Reply:
I've been using a little CSharp app for testing - I've not found anything in JS.